app/src/features/projects/ProjectPage.tsx

```diff
@@ -48,9 +69,69 @@ const ProjectPage: React.FC = () => {
 
   const biohubApi = useBiohubApi();
 
+  const classes = useStyles();
+
+  const [files, setFiles] = useState<File[]>([]);
+  const [dropzoneText, setDropzoneText] = useState<string>('Select files');
+  const [dropzoneInstanceKey, setDropzoneInstanceKey] = useState<number>(0);
+
+  // Whether or not to show the text dialog
+  const [openErrorDialogProps, setOpenErrorDialogProps] = useState<IErrorDialogProps>({
+    dialogTitle: UploadProjectArtifactsI18N.uploadErrorTitle,
+    dialogText: UploadProjectArtifactsI18N.uploadErrorText,
+    open: false,
+    onClose: () => {
+      setOpenErrorDialogProps({ ...openErrorDialogProps, open: false });
+    },
+    onOk: () => {
+      setOpenErrorDialogProps({ ...openErrorDialogProps, open: false });
+    }
+  });
+
   // TODO this is using IProject in the mean time, but will eventually need something like IProjectRecord
   const [project, setProject] = useState<IProject | null>(null);
 
+  const handleUpload = async () => {
+    const showErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
+      setOpenErrorDialogProps({ ...openErrorDialogProps, ...textDialogProps, open: true });
+    };
+
+    if (files && files.length > 0) {
```

Comment:
A pattern that I've seen recommended online, and which I also quite like, is instead of something like this:

```Typescript
if(happy) {
 ...happyPath
} else {
 ...sadPath
}
```
Instead doing:
```Typescript
if(!happy) {
 ...sadPath
 return;
}

...happyPath
```

It eliminates some of the complexity by reducing if/else complexity (especially if the happyPath has its own if/elses.  And it reads a bit easier because the developer can read top->bottom, and if anything is incorrect the code returns immediately.  You don't have to skip over happy path code to get to the sad path code.

Reply:
just added a commit to fix .. let me know what you think